honeybee_energy/ventcool/crack.py

```diff
@@ -0,0 +1,165 @@
+# coding=utf-8
+"""Definition of surface crack for multizone airflow."""
+from __future__ import division
+
+from honeybee._lockable import lockable
+
+
+@lockable
+class AFNCrack(object):
+    """Definition of surface crack for multizone airflow.
+
+    Args:
+        air_mass_flow_coefficient_reference: A number in kg/s-m at 1 Pa per meter of
+            crack length at the conditions defined in the ReferenceCrack condition;
+            required to run an AirflowNetwork simulation. Some common values for this
+            coefficient from the DesignBuilder Cracks template include the following:
+
+                * 0.00001 - Tight low-leakage external wall
+                * 0.001 - Tight, low-leakage internal wall
+                * 0.0004 - Poor, high-leakage external wall
+                * 0.019 - Poor, high-leakage internal wall
+
+        air_mass_flow_exponent: An optional dimensionless number between 0.5 and 1 used
+            to calculate the crack mass flow rate; required to run an AirflowNetwork
+            simulation. This value represents the leak geometry impact on airflow, with
+            0.5 generally corresponding to turbulent orifice flow and 1 generally
+            corresponding to laminar flow. The default of 0.65 is representative of many
+            cases of wall and window leakage, used when the exponent cannot be measured.
+            (Default: 0.65).
+        crack_factor: A number indicating multiplier for air mass flow through a crack.
+            (Default: 1).
+
+    Properties:
+        * air_mass_flow_coefficient_reference
+        * air_mass_flow_exponent
+        * crack_factor
+        * parent
+        * has_parent
+    """
+    __slots__ = ('_air_mass_flow_coefficient_reference', '_air_mass_flow_exponent',
+                 '_crack_factor', '_parent', '_locked')
+
+    def __init__(self, air_mass_flow_coefficient_reference, air_mass_flow_exponent=0.65,
+                 crack_factor=1):
+        """Initialize AFNCrack."""
+        self.air_mass_flow_coefficient_reference = air_mass_flow_coefficient_reference
+        self.air_mass_flow_exponent = air_mass_flow_exponent
+        self.crack_factor = crack_factor
+        self._parent = None  # this will be set when assigned to an aperture
+
+    @property
+    def air_mass_flow_coefficient_reference(self):
+        """Get or set the air mass flow coefficient defined at the reference crack."""
+        return self._air_mass_flow_coefficient_reference
+
+    @air_mass_flow_coefficient_reference.setter
+    def air_mass_flow_coefficient_reference(self, value):
+        assert value > 0, 'The air_mass_flow_coefficient_reference must be greater ' \
+            'than 0. Got: {}.'.format(value)
```

Comment:
This isn't the right way to do range checks in honeybee since there are a bunch of other cases where people input strings, etc. See [how it's done in the VentilationOpening class here](https://github.com/ladybug-tools/honeybee-energy/blob/master/honeybee_energy/ventcool/opening.py#L66).

Reply:
Or, in this case, you probably want to use the `float_positive` check that is in the same honeybee-core module.